plugins/defaultplugins/ifplugin/stn_config.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ifplugin
+
+import (
+	"fmt"
+
+	govppapi "git.fd.io/govpp.git/api"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/measure"
+	"github.com/ligato/cn-infra/utils/addrs"
+	"github.com/ligato/cn-infra/utils/safeclose"
+	"github.com/ligato/vpp-agent/idxvpp"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/bin_api/stn"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
+	model_stn "github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/model/stn"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/vppcalls"
+	"github.com/ligato/vpp-agent/plugins/govppmux"
+	"github.com/prometheus/common/route"
+)
+
+type StnConfigurator struct {
+	Log logging.Logger
+
+	GoVppmux    govppmux.API
+	StnIndexes  idxvpp.NameToIdxRW
+	StnIndexSeq uint32
+	SwIfIndexes ifaceidx.SwIfIndex
+	vppChan     *govppapi.Channel
+
+	Stopwatch *measure.Stopwatch
+}
+
+// Init initializes ARP configurator
+func (plugin *StnConfigurator) Init() (err error) {
+	plugin.Log.Debug("Initializing StnConfigurator")
+
+	// Init VPP API channel
+	plugin.vppChan, err = plugin.GoVppmux.NewAPIChannel()
+	if err != nil {
+		return err
+	}
+
+	err = plugin.checkMsgCompatibility()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Check STN rule raw data
+func CheckStn(stnInput *model_stn.StnRule, index ifaceidx.SwIfIndex, log logging.Logger) (*vppcalls.StnRule, error) {
+	if stnInput == nil {
+		log.Infof("STN input is empty")
+		return nil, nil
+	}
+	if stnInput.Interface == "" {
```

Comment:
If the Interface and the IpAddress are required during both, Add and Delete, do not log it as an Info but return an error. The rule won't be configured in such a case which is an unwanted state.